index.js

```diff
@@ -0,0 +1,365 @@
+'use strict';
+
+const Scm = require('screwdriver-scm-base');
+const async = require('async');
+const hoek = require('hoek');
+
+process.on('unhandledRejection', console.dir);
+
+class ScmRouter extends Scm {
+    /**
+     * Constructs a router for different scm strategies
+     * @method constructor
+     * @param  {Object}         config                             Object with scms and ecosystem
+     * @param  {Object}         [config.ecosystem]                 Optional object with ecosystem values
+     * @param  {Object}         config.scms                        Array of scms to load or a single scm object
+     * @param  {String}         config.scms[x].plugin              Name of the scm NPM module to load
+     * @param  {String}         config.scms[x].config              Configuration to construct the module with
+     * @param  {String}         config.scms[x].config.displayName  Nickname to displaoy of the scm
+     * @return {ScmRouter}
+     */
+    constructor(config = {}) {
+        const ecosystem = config.ecosystem;
+        const scmsConfig = config.scms;
+
+        super();
+
+        this.scms = {};
+
+        if (typeof scmsConfig === 'object') {
+            if (!(Array.isArray(scmsConfig))) {
+                throw new Error('No scm config passed in.');
+            }
+            scmsConfig.forEach((scm) => {
+                if (scm.config.displayName && typeof scm.config.displayName === 'string') {
+                    const options = hoek.applyToDefaults({ ecosystem },
+                        (scm.config || {}));  // Add ecosystem to scm options
```

Comment:
why we need `(scm.config || {})`? The `if` condition has checked that `scm.config.displayName` exists